py/visdom/server.py

```diff
@@ -928,15 +931,50 @@ def initialize(self, app):
         self.state = app.state
         self.port = app.port
         self.env_path = app.env_path
+        self.with_login = app.with_login
+        self.user_login = app.user_login
 
     def get(self, args, **kwargs):
         items = gather_envs(self.state, env_path=self.env_path)
-        self.render(
-            'index.html',
-            user=getpass.getuser(),
-            items=items,
-            active_item=''
-        )
+        if self.with_login == False:
```

Comment:
Can you please merge this logic?

```
if (not self.with_login) or (self.with_login and self.current_user):
  # render index.html
else:
  # render login.html
```

Note that I used `self.current_user`, which is described here: http://www.tornadoweb.org/en/stable/guide/security.html#user-authentication